soc/xtensa/intel_adsp/tools/cavstool.py

```diff
@@ -856,7 +860,13 @@ def get_host_ip():
 ap.add_argument("-n", "--no-history", action="store_true",
                 help="No current log buffer at start, just new output")
 ap.add_argument("-s", "--server-addr",
-                help="Specify the IP address that the server to active")
+                help="Specify the IP address that the log server to active")
+ap.add_argument("-i", "--interface",
+                help="Specify the network interface where the service runs")
```

Comment:
```suggestion
                help="Specify the network interface the service will LISTEN to")
```
A service never runs on a specific network interface. It can BIND to a single network interface to LISTEN to when it starts (not to random one by default, this is the current bug) and then the service completely forgets it did that.

Reply:
> A service never runs on a specific network interface. It can BIND to a single network interface to LISTEN to when it starts (not to random one by default, this is the current bug) and then the service completely forgets it did that.

Thank you! @marc-hb , how do you think if we make the user must select the network interface while there are more than one network interface existing (except localhost)?

Reply:
> Thank you! @marc-hb , how do you think if we make the user must select the network interface while there are more than one network interface existing (except localhost)?

I think we already discussed this. The fix is to bind to `INADDR_ANY` (= all interfaces) by default when nothing is provided on the command line. That's it.